docs-ref-conceptual/azure-cli-extension-alias.md

```diff
@@ -0,0 +1,149 @@
+
+```azurecli
+[rg]
+command = group
+
+[ls]
+command = list
+```
+
+These newly defined aliases can now be used anywhere that their definition would be.
+
+```azurecli
+az rg list
+az rg ls
+az vm ls
+```
+
+Aliases can also be shortcuts for complete commands. For example, here is an alias named  that lists available resource groups and their locations as an ASCII table:
+
+```
+[lsgroups]
+command = group list --query [].{Name:name, Location:location} --output table
+```
+
+Now `lsgroups` can be run like any other CLI command.
+
+```azurecli
+az lsgroups
+```
+
+This type of alias is most useful when you have a command you run often and want a short name for it.
+
+## Create an alias command with arguments
+
+You can also add positional arguments to an alias command, by including them as `{{ arg_name }}` in the alias name. The whitespace in the curly braces is required.
+
+```
+[alias_name {{ arg1 }} {{ arg2 }} ...]
+command = invoke_including_args
+```
+
+You will be working with these types of aliases frequently to take advantage of brevity while still allowing arguments to be specified. These arguments can appear anywhere in the invoked command, and don't have to correspond to arguments of the original command.
+
+In the following command, the name of a resource group and VM are used and the first available public IP address of the VM is returned.
+
+```
+[get-vm-ip {{ resourceGroup }} {{ vmName }}]
+command = vm list-ip-addresses --resource-group {{ resourceGroup } --name {{ vmName }} --query [0].virtualMachine.network.publicIpAddresses[0].ipAddress
+```
+
+When running this command, you give values to the positional arguments. This example gets the IP information for the vm `MyVM` in the `MyResourceGroup` resource group.
+
+```azruecli
+az get-vm-ip MyResourceGroup MyVM
+```
+
+You can also use environment variables in commands invoked by aliases. These environment variables are evaluated at runtime. The next example adds the `create-rgrp` alias, which creates a new resource group in `eastus` and adds an `owner` tag with the value of the local `USER` environment variable.
+
+```
+[create-rgrp {{ groupName }}]
+command = group create --name {{ groupName }} --location eastus --tags owner=$USER
+```
+
+## Process arguments using Jinja2 templates
+
+Argument substitution in the alias extension is performed by [Jinja2](http://jinja.pocoo.org/docs/2.10/) and so when
+expanding a positional argument, you have access to the full capabilities of the jinja template engine. Templates allow you to perform actions like data extraction and substitution on strings. The next example uses the functionality of Jinja2 templates to create an alias command that takes a storage URL and extracts the account and container names.
+
+```
+[storage-ls {{ url }}]
+command = storage blob list --account-name {{ url.replace('https://', '').split('.')[0] }} --container-name {{ url.replace('https://', '').split('/')[1] }}
```

Comment:
The Jinja2 template engine requires `{% ... %}` as the delimiter for statements (and `{{ ... }}` for expressions, as used elsewhere.) Does this need to change here? cc @derekbekoe

Reply:
@troydai or @chewong would know.

Reply:
Technically, users can use `{% ... %}` in their alias configuration file to perform actions such as if statements but the primary use case of the Jinja template engine is to incorporate positional arguments and additional string processing like replace or split. So I think `{{ ... }}` would be sufficient for most of the use case.